### PR TITLE
Update old qunit test - integration_test.js

### PR DESCRIPTION
**Pull request by Alex7Li**
This commit is built on commit #21. the diff will be wrong until it's merged. 

This commit is part of an effort to migrate old tests to karma. As this is moving the integration tests, there are some changes of note. Karma needs to use two different configurations for the integration and unit tests, so I created a different karma.conf.js file just for running the integration tests in test/integration. Since it would be annoying to have to run both every time, I also set up a grunt task to run them.

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -50,17 +50,30 @@ module.exports = function(grunt) {
         },
       },
     },
-    'qunit': {
-      files: ['test/unit.html', 'test/integration.html'],
+    qunit: {
```
Comment (kjgalvan) [inline Gruntfile.js:53]: NIT: We should have this be consistent and either wrap attributes in single quotes or not at all

File: test/integration/karma.conf.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+const baseConfig = require('../../karma.conf');
+
+module.exports = function(config) {
+  baseConfig(config);
+  config.set({
+    files: [
+      './**_test.js',
```
Comment (kjgalvan) [inline test/integration/karma.conf.js:7]: You may want to explicitly bring in only the integration test as future tests will have goog.module() syntax which may cause issues for Karma running this config.

File: Gruntfile.js (hunk 1 above)
Comment (Alex7Li) [reply Gruntfile.js:53]: Fixed

File: test/integration/karma.conf.js (hunk 2 above)
Comment (Alex7Li) [reply test/integration/karma.conf.js:7]: Actually, this does only bring in the integration tests. The file paths are relative, and this file is in the 'test/integration' directory, so it should only grab test files placed in the integration directory.

File: test/integration/integration_test.js [hunk 3]
```diff
@@ -0,0 +1,207 @@
+describe('The data layer helper library', function() {
+  it('does not pollute the global scope', function() {
+    expect(window['dataLayer']).toBe(undefined);
```
Comment (zikas) [inline test/integration/integration_test.js:3]: Could you please take a look to migration of these tests to Jasmine's toBeUndefined()/spy() etc?

File: README.md [hunk 4]
```diff
@@ -471,12 +471,14 @@ To run the new tests, you will need to first install karma globally:
...
 ```
 
-You can run the new tests with the command
+You can run tests with the command
 
 ```bash
 karma start
 ```
 
+If called in the top level ./ directory, the unit tests will run. If called
```
Comment (zikas) [inline README.md:480]: This is a bit confusing.. can we change it so it's controlled by a flag and should always be run from the root folder?

File: test/integration/integration_test.js (hunk 3 above)
Comment (Alex7Li) [reply test/integration/integration_test.js:3]: I will put it in a different commit

File: README.md (hunk 4 above)
Comment (Alex7Li) [reply README.md]: I found a way to run from root with karma. Also, if you run the grunt command from root, it runs the integration tests and the unit tests by default. Updated the README with this information.

**Pull request merged**